app/src/main/java/io/pslab/models/PSLabSensor.java

```diff
@@ -316,9 +318,12 @@ public boolean onOptionsItemSelected(MenuItem item) {
               will fire up. If user declines to give permission, don't do anything.
              */
             case R.id.record_data:
-                if (!isRecording) {
+                if (!isRecording && (!sensorNotFound() || scienceLab.isConnected())) {
```

Comment:
According to this logic, if PSLab is connected, it doesn't check the sensor test right?

Reply:
It checks but user would be able to start the recording nevertheless since PSLab is connected.